core/sail/memory/src/main/java/org/eclipse/rdf4j/sail/memory/FileIO.java

```diff
@@ -306,6 +314,9 @@ private Value readValue(DataInputStream dataIn) throws IOException, ClassCastExc
 			String label = readString(dataIn);
 			IRI datatype = (IRI) readValue(dataIn);
 			return vf.createLiteral(label, datatype);
+		} else if (valueTypeMarker == RDFSTAR_TRIPLE_MARKER) {
+			IRI rdfStarEncodedTriple = (IRI) readValue(dataIn);
+			return RDFStarUtil.fromRDFEncodedValue(rdfStarEncodedTriple);
```

Comment:
Do any of the tests check that this code works by adding data, shutting down the repo, then creating a new one with the same directory and check if all the data is still there?

Reply:
I _think_ we cover that in some of the integration tests, but I'm not 100% sure to be honest. I was also a bit surprised the FileIO itself has no direct unit test coverage. Might be worth putting a followup card in for that.

Reply:
Added a card to look into this further: https://github.com/eclipse/rdf4j/issues/3771